includes/admin/donors/donor-functions.php

```diff
@@ -165,3 +165,60 @@ function give_connect_user_donor_profile( $donor, $donor_data, $address ) {
 
 	return $output;
 }
+
+/**
+ * Delete Donor using Bulk Actions.
+ *
+ * @param int   $donor_id Donor ID.
+ * @param array $args     An Array of Additional Donor Arguments.
+ *
+ * @since 1.8.17
+ *
+ * @return bool
+ */
+function give_delete_bulk_donors( $donor_id, $args ) {
+	$donor_edit_role = apply_filters( 'give_edit_donors_role', 'edit_give_payments' );
+
+	if ( ! is_admin() || ! current_user_can( $donor_edit_role ) ) {
+		wp_die( __( 'You do not have permission to delete donors.', 'give' ), __( 'Error', 'give' ), array(
+			'response' => 403,
+		) );
+	}
+
+	// Bail Out, if Donor ID doesn't exists.
+	if ( empty( $donor_id ) ) {
+		return false;
+	}
+
+	$nonce = $args['_wpnonce'];
+
+	// Verify Nonce for deleting bulk donors.
+	if ( ! wp_verify_nonce( $nonce, 'delete-bulk-donors' ) ) {
+		wp_die( __( 'Cheatin&#8217; uh?', 'give' ), __( 'Error', 'give' ), array(
+			'response' => 400,
+		) );
+	}
+
+	$donor = new Give_Donor( $donor_id );
+
+	if ( $donor->id > 0 ) {
+		$payments_ids  = explode( ',', $donor->payment_ids );
+		$donor_deleted = Give()->donors->delete( $donor->id );
+
+		if ( $donor_deleted ) {
+
+			// Remove all donations, logs, etc.
+			foreach ( $payments_ids as $payment_id ) {
+				give_delete_donation( $payment_id );
+			}
+		} else {
```

Comment:
@mehul0810 Please show an error to admin him somehow donor does not delete instead of removing a donor from donations.